thavalon-webapp/src/components/gameCreation.tsx

```diff
@@ -0,0 +1,326 @@
+    displayName: string
+}
+
+/**
+ * Required data to create a game.
+ */
+interface CreateGameData {
+    displayName: string
+}
+
+/**
+ * Props for both the Join and Create Game forms
+ */
+interface FormProps {
+    showForm: Dispatch<SetStateAction<boolean>>,
+    onSuccess: () => void
+}
+
+/**
+ * Enum representing the current modal state. Used to switch between the different forms.
+ */
+enum CreateJoinState {
+    CreateJoinButtons,
+    CreateGame,
+    JoinGame
+}
+
+const connection = AccountManager.getInstance();
+
+/**
+ * Creates the modal for the join and create game options. This modal appears
+ * when a user clicks "Play" and can either be used to create join a game.
+ * @param props - Required reqct props object.
+ */
+export function CreateJoinGameModal(props: CreateJoinGameProps): JSX.Element {
+    // State to keep track of whether or not the form should display.
+    // Needed to trigger closing animations when the form is leaving the modal.
+    const [showForm, setShowForm] = useState(false);
+    // State to track the overall modal state, including which form should be displayed.
+    const [modalState, setModalState] = useState(CreateJoinState.CreateJoinButtons);
+
+    /**
+     * Function handler for the success event from the child modal.
+     * Will reset state for future callers and close the modal.
+     */
+    function onSuccess(): void {
+        setModalState(CreateJoinState.CreateJoinButtons);
+        props.setOpen(false);
+    }
+
+    return (
+        <ReactModal
+            isOpen={props.show}
+            onRequestClose={() => props.setOpen(false)}
+            contentLabel="Create/Join Game Modal"
+            className="Modal"
+            overlayClassName="Overlay">
+            <div className="modalContainer">
+                <h2 className="modalHeader">Play Thavalon</h2>
+                <hr />
+                {modalState === CreateJoinState.CreateJoinButtons && CreateJoinButtons(setShowForm, setModalState)}
+                <CSSTransition
+                    in={showForm}
+                    timeout={300}
+                    classNames="create-join-form"
+                    onExited={() => setModalState(CreateJoinState.CreateJoinButtons)}
+                    unmountOnExit>
+                    <>
+                        {modalState === CreateJoinState.CreateGame && <CreateGameForm showForm={setShowForm} onSuccess={onSuccess} />}
+                        {modalState === CreateJoinState.JoinGame && <JoinGameForm showForm={setShowForm} onSuccess={onSuccess} />}
+                    </>
+                </CSSTransition>
+            </div>
+        </ReactModal>
+    );
+}
+
+/**
+ * 
+ * @param showForm Sets whether or not a form should display
+ * @param setModalState Sets the modal state to the correct form that should display
+ */
+function CreateJoinButtons(showForm: Dispatch<SetStateAction<boolean>>, setModalState: Dispatch<SetStateAction<CreateJoinState>>): JSX.Element {
```

Comment:
Up to you, but should each function be there own file? That way, if we ever want to reuse them it's easier, and we have the idea that one tsx file = 1 component

Reply:
My reasoning for keeping this in the same module is that they are very specific to this component. In particular, things like the class names, the labels, and the end points are all only used by this component. I only split them to prevent one giant component and improve readability. I also think this still conforms to the one tsx file = 1 public component,